Sources/HaCWebsiteLib/Events/GeneralEvent.swift

```diff
@@ -1,44 +1,125 @@
 import Foundation
+import Fluent
 
-struct GeneralEvent: PostCardRepresentable {
-    let title : String
-    let time : DateInterval
-    let tagLine : String
-    let color : String
-    let hypePeriod : DateInterval
-    let tags : [String]
-    let eventDescription : Markdown
-    let imageURL : String? 
-    let location : Location?
-    let facebookEventID : String?
-    var shouldShowAsUpdate : Bool {
-        get {
-            return self.hypePeriod.contains(Date())
-        }
-    }
-    var postCardRepresentation : PostCard {
-        return PostCard(
-          title: self.title,
-          category: .general,
-          description: self.tagLine,
-          backgroundColor: self.color, //TODO
-          imageURL: self.imageURL
-        )
+final class GeneralEvent: Event, Entity {
+  //This is missing a workshop attribute from the deprecated WorkshopEvent.swift
+  let storage = Storage()
+  let title : String
+  let time : DateInterval
+  let tagLine : String
+  let color : String
+  let hypePeriod : DateInterval
+  let tags : [String]
+  let eventDescription : Markdown
+  let websiteURL : String?
+  let imageURL : String? 
+  let location : Location?
+  let facebookEventID : String?
+  var shouldShowAsUpdate : Bool {
+    get {
+      return self.hypePeriod.contains(Date())
+    }   
+  }
+  var postCardRepresentation : PostCard {
+    return PostCard(
+        title: self.title,
+        category: .general,
+        description: self.tagLine,
+        backgroundColor: self.color, //TODO
```

Comment:
What's this TODO referring to?

Reply:
@Pinpickle this is from before when I think we were floating the idea of not using strings for colors. Maybe this deserves to be an Issue instead?

Reply:
Yeah, sounds like a plan